### PR TITLE
execution/engineapi: switch `baseFeePerGas` and `blockValue` to `hexutil.U256`

### PR DESCRIPTION
**Pull request by AskAlexSharov**
problem: the Engine API types carry `baseFeePerGas` and `blockValue` as `*hexutil.Big`, so newPayload and getPayload convert `uint256` -> `big.Int` -> `uint256` for every payload.

Solution: carry them as `*hexutil.U256`. The JSON is byte-identical and `hexutil.U256` accepts and rejects the same inputs as `hexutil.Big`; only the Go type name in decode error messages changes. `TransitionConfiguration` had no users and is removed.

n1, getPayload response with 16 transactions, 6 runs: decode 35 -> 33 allocs, encode 40 -> 36 allocs; time within noise.

### CODE REVIEW

File: execution/engineapi/engine_types/jsonrpc.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ type ExecutionPayload struct {
 	GasUsed         hexutil.Uint64        `json:"gasUsed"`
 	Timestamp       hexutil.Uint64        `json:"timestamp"`
 	ExtraData       hexutil.Bytes         `json:"extraData"`
-	BaseFeePerGas   *hexutil.Big          `json:"baseFeePerGas"`
+	BaseFeePerGas   *hexutil.U256         `json:"baseFeePerGas"`
```
[the remark this answers is not in the document]
Comment (AskAlexSharov) [reply execution/engineapi/engine_types/jsonrpc.go:49]: Fixed the base fee in 85576fe33a: it is SSZ-encoded from the U256 directly. The block value keeps one ToInt(), because GetAssembledBlock returns *big.Int to Caplin's block production; changing that interface is out of scope here.

File: execution/engineapi/engine_server.go [hunk 2]
```diff
@@ -326,7 +326,7 @@ func (s *EngineServer) newPayload(ctx context.Context, req *engine_types.Executi
 		Coinbase:    req.FeeRecipient,
 		Root:        req.StateRoot,
 		Bloom:       bloom,
-		BaseFee:     uint256.MustFromBig(req.BaseFeePerGas.ToInt()),
+		BaseFee:     new(uint256.Int).Set((*uint256.Int)(req.BaseFeePerGas)),
```
Comment (awskii) [inline execution/engineapi/engine_server.go]: **newPayload panics on a null or missing baseFeePerGas (regression from the U256 migration)**

`BaseFee: new(uint256.Int).Set((*uint256.Int)(req.BaseFeePerGas))` at engine_server.go:329 dereferences `req.BaseFeePerGas` without a nil check, because `uint256.Int.Set` reads `x[0..3]`.

A nil value can reach this line:
- `ExecutionPayload` has no custom JSON decoder, and `encoding/json` leaves a pointer field nil when the field is `null` or absent. The PR's own `TestGetPayloadResponseQuantitiesJSON` decodes `null` quantities without error.
- Nothing before line 329 checks `BaseFeePerGas`.

This PR introduced it. The old line `uint256.MustFromBig(req.BaseFeePerGas.ToInt())` was nil-safe:
- `(*hexutil.Big).ToInt` is a plain cast with no dereference.
- `MustFromBig(nil)` returns nil (uint256 `conversion.go:123-125`).
- The header therefore had no BaseFee, and the request went through normal validation instead of panicking.

Consequence: the RPC callback wrapper (`rpc/service.go:228-233`) recovers the panic. It logs `RPC method ... crashed` with a stack trace at ERROR and returns the generic `method handler crashed` error instead of a PayloadStatus or `-32602`. The node keeps running, and the caller is the JWT-authenticated CL, so the severity is minor rather than major.

The SSZ-REST and Caplin paths are unaffected, because `ExecutionPayloadFromSSZBlock` always sets a non-nil base fee. The same unguarded `Set` is in `engineapitester/mock_cl.go:434`; it is only a latent hazard there, because getPayload always sets the base fee.

Fix: Check `req.BaseFeePerGas` for nil before building the header and return `&rpc.InvalidParamsError{Message: "missing baseFeePerGas"}`, or keep nil as nil as the old code did. Apply the same guard in `mock_cl.go`, and add a newPayload test that sends a null baseFeePerGas and expects a proper error instead of a panic.

**Review by awskii — APPROVED** [review]
One minor point inline.

File: execution/engineapi/engine_server.go (hunk 2 above)
Comment (AskAlexSharov) [reply execution/engineapi/engine_server.go]: Fixed in 6e1cf49927: a null baseFeePerGas stays nil, as before, in newPayload and in mock_cl; TestNewPayloadWithoutBaseFeeDoesNotPanic panicked at engine_server.go:329 before the fix.
